### PR TITLE
T1s2 form and validation

### PR DESCRIPTION
**Pull request by Joperes-ym**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -56,3 +55,23 @@ function displayCard(array $cards) : string
     }
     return $result;
 }
+
+function allDataPresent(array $userData): bool
```
Comment (Steph-ski) [inline functions.php:65]: Add a dock block for this function

File: functions.php [hunk 2]
```diff
@@ -56,3 +55,23 @@ function displayCard(array $cards) : string
...
+{
+    return (isset($userData['name']) && isset($userData['type_line']) && isset($userData['mana']) && isset($userData['rarity']) && isset($userData['img_url']));
+}
+
+// validation
```
Comment (Steph-ski) [inline functions.php:70]: Add a dock block for this function

File: functions.php [hunk 3]
```diff
@@ -56,3 +55,23 @@ function displayCard(array $cards) : string
...
+function validLength(string $data): bool
+{
+    return !(strlen($data) > 100);
+}
+
+function validRarity(string $rarity): bool
```
Comment (Steph-ski) [inline functions.php:86]: Add a dock block for this function

File: index.php [hunk 4]
```diff
@@ -22,6 +24,7 @@
         <div>
```
Comment (Steph-ski) [inline index.php:24]: I ran the HTML through the validator and it picked up some errors. Would recommend running through HTML validator and check with Charlie if they are strict errors which need sorting (as PHP code is being prioritised) .

File: functions.php [hunk 5]
```diff
@@ -56,3 +55,23 @@ function displayCard(array $cards) : string
...
     return $result;
 }
+
+function allDataPresent(array $userData): bool
+{
+    return (isset($userData['name']) && isset($userData['type_line']) && isset($userData['mana']) && isset($userData['rarity']) && isset($userData['img_url']));
+}
+
+// validation
+function validLength(string $data): bool
```
Comment (Steph-ski) [inline functions.php:76]: Just a suggestion, may want to add in something to not allow special characters as well as checking length.

**Review by Steph-ski — COMMENTED** [review]
Page looks really good. 

Unit test all passed.

Submitted valid form, however I input special characters and it still validated - you might want to include this in your validation, just a suggestion. 

Couple of HTML errors on validator but again not sure how pressing these are - may want to check with Charlie. 

Testing file in Testing Directory is called indexTests, may be better to rename this FunctionsTest?

File: functions.php (hunk 5 above)
Comment (Joperes-ym) [reply functions.php:76]: Nicely noted. Just not sure if I'll have the time :(

**Review by Steph-ski — APPROVED** [review]

**Pull request merged**